src/dawn/Optimizer/PassStageSplitter.cpp

```diff
@@ -28,6 +28,19 @@
 
 namespace dawn {
 
+namespace {
+
+static void reportDataDependencyInsideConditionalBlock(
+    const std::shared_ptr<Statement>& statement,
+    const std::shared_ptr<iir::StencilInstantiation>& instantiation) {
+
+  DiagnosticsBuilder diag(DiagnosticsKind::Error, statement->ASTStmt->getSourceLocation());
```

Comment:
does it make sense to pack a 3-liner into a function that is only called once?

Reply:
I was mirroring the same functional semantics separation that is also employed here:
https://github.com/MeteoSwiss-APN/dawn/blob/9513f9723c597042b1ca02d02280f8b7f9790514/src/dawn/Optimizer/PassFieldVersioning.cpp#L68-L90
Maybe more verbose diagnostic could be required and it would be ideal then to have already a function for the reporting purpose only.

Reply:
The difference is that this function is called more than once and it would have duplicated code to inline it whereas your call is only needed in one place.

Reply:
Ok, I'm moving the code to the `run` method on the next commit.